test/backend/lib.api.socket-handler.test.js

```diff
@@ -13,15 +13,27 @@ var express = require('express');
 var https = require('https');
 var path = require('path');
 var fs = require('fs');
+var readLineSync = require('readline-sync')
 
 function createServer(app) {
   // Get full path for certificate files
   var keyFile = path.resolve(__dirname, '../../config/key.pem');
   var certFile = path.resolve(__dirname, '../../config/cert.pem');
-  return https.createServer({
-    key: fs.readFileSync(keyFile),
-    cert: fs.readFileSync(certFile)
-  }, app);
+  try {
+    // No passphrase when cert is generated with -nodes
+    server = require('https').createServer({
+      key: fs.readFileSync(keyFile),
+      cert: fs.readFileSync(certFile),
+    }, app);
+  } catch {
+    // Prompts for passphrase
+    server = require('https').createServer({
+      key: fs.readFileSync(keyFile),
+      cert: fs.readFileSync(certFile),
+      passphrase: readLineSync.question("Enter PEM passphrase: ")
+    }, app);
+  }
```

Comment:
This seems like a nice chunk of duplication that could be DRYed up

Reply:
Absolutely.